doc/migration.md

```diff
@@ -0,0 +1,158 @@
+# Package changes
+
+Classes in the library have been re-organised to simplify the packaging.
+
+All classes are now in the `com.cloudant.sync` package:
+
+* All private API classes are in the `com.cloudant.sync.internal`
+  package. API users are discouraged from using these classes as fields,
+  method signatures, and implementation details may be subject to
+  change. These classes have the same status as those marked "API
+  Status: Private" in the javadoc for the 1.x versions of the library.
+* Everything else under `com.cloudant.sync` is public API and subject
+  to the usual versioning and deprecation practices. API users can
+  expect this to be a stable API.
+* Classes in the `com.cloudant.sync.internal` package are not visible
+  in javadoc.
+
+API users will have to adjust their `import` statements
+appropriately. Furthermore, some class names and methods have also
+changed (see below).
+
+# The `Datastore` has been replaced with the `DocumentStore` class
+
+The `DatastoreManager` class has been removed. Replace instances of
+
+```java
+File path = getApplicationContext().getDir("datastores"); // Android-specific
+DatastoreManager manager = DatastoreManager.getInstance(path.getAbsolutePath());
+Datastore ds = manager.openDatastore("my_datastore");
+// read a doc
+DocumentRevision dr = ds.getDocument("my-document-id");
+```
+
+with
+
+```java
+File path = getApplicationContext().getDir("datastores");
+DocumentStore ds = DocumentStore.getInstance(new File(path, "my_datastore"));
+// read a doc
+DocumentRevision dr = ds.database().read("my-document-id");
+```
+
+As the above example shows, "CRUD" (create, read, update, delete)
+functionality has been migrated to the new `Database` class. Obtain an
+instance of the `Database` class managed by the `Datastore` by calling
+the `database()` getter method.
+
+The methods on the `Database` methods are different to their
+counterparts on `Datastore`:
+
+* `getDocument` has been renamed `read`
+* `getAllDocuments` has been renamed `read`
+* `getDocumentsWithIds` has been renamed `read`
+* `containsDocument` has been renamed `contains`
+* `getAllDocumentIds` has been renamed `getIds`
+* `getConflictedDocumentIds` has been renamed `getConflictedIds`
+* `resolveConflictsForDocument` has been renamed `resolveConflicts`
+* `createDocumentFromRevision` has been renamed `create`
+* `updateDocumentFromRevision` has been renamed `update`
+* `deleteDocumentFromRevision` has been renamed `delete`
+* `deleteDocument` has been renamed `delete`
+
+The `throws` clauses on `Database` methods are different to those on
+their counterparts in `Datastore`. Code which calls these methods may
+have to be adjusted to catch different exceptions.
```

Comment:
It's probably worth another section on `Exceptions` that could be linked from here and query that says something along the lines of how we've made an effort to use checked exceptions instead of runtime exceptions to reduce the likelihood of unexpected application crashes. Maybe worth saying that runtime exceptions may still be thrown in some circumstances and these have been documented where possible.